diff --git a/MODULE/Makefile b/MODULE/Makefile
index 288d813..33cbd5c 100755
--- a/MODULE/Makefile
+++ b/MODULE/Makefile
@@ -15,7 +15,7 @@ endif
 RTMP_SRC_DIR = $(RT28xx_DIR)/RT$(CHIPSET)
 
 #PLATFORM: Target platform
-PLATFORM = PC
+#PLATFORM = PC
 #PLATFORM = 5VT
 #PLATFORM = IKANOS_V160
 #PLATFORM = IKANOS_V180
@@ -37,6 +37,7 @@ PLATFORM = PC
 #PLATFORM = KODAK_DC
 #PLATFORM = DM6446
 #PLATFORM = FREESCALE8377
+PLATFORM = SONIX
 
 #RELEASE Package
 RELEASE = DPB
@@ -53,6 +54,11 @@ ifeq ($(TARGET),THREADX)
 MAKE = gmake
 endif
 
+ifeq ($(PLATFORM), SONIX)
+LINUX_SRC = /home/szricky/work/sdk/snx_sdk/kernel/linux-2.6.35.12/src
+CROSS_COMPILE = arm-linux-
+endif
+
 ifeq ($(PLATFORM),5VT)
 LINUX_SRC = /project/stable/5vt/ralink-2860-sdk/linux-2.6.17
 CROSS_COMPILE = /opt/crosstool/uClibc_v5te_le_gcc_4_1_1/bin/arm-linux-
@@ -223,9 +229,9 @@ ifeq ($(PLATFORM),INF_AMAZON_SE)
 endif
 else	
 ifeq ($(RT28xx_MODE),APSTA)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)apsta.o /tftpboot
+#	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)apsta.o /tftpboot
 else
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)sta.o /tftpboot
+#	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)sta.o /tftpboot
 endif	
 endif	
 else
@@ -242,14 +248,14 @@ endif
 endif
 
 ifeq ($(RT28xx_MODE),AP)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)ap.ko /tftpboot
+	#cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)ap.ko /tftpboot
 #	rm -f os/linux/rt2860ap.ko.lzma
 #	/root/bin/lzma e os/linux/rt$(CHIPSET)ap.ko os/linux/rt$(CHIPSET)ap.ko.lzma
 else	
 ifeq ($(RT28xx_MODE),APSTA)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)apsta.ko /tftpboot
+	#cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)apsta.ko /tftpboot
 else
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)sta.ko /tftpboot
+	#cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)sta.ko /tftpboot
 endif
 endif	
 endif
diff --git a/MODULE/os/linux/config.mk b/MODULE/os/linux/config.mk
index 98f0c8f..9c759bb 100755
--- a/MODULE/os/linux/config.mk
+++ b/MODULE/os/linux/config.mk
@@ -565,3 +565,8 @@ ifeq ($(PLATFORM),DM6446)
 	CFLAGS := -nostdinc -iwithprefix include -D__KERNEL__ -I$(RT28xx_DIR)/include -I$(LINUX_SRC)/include  -Wall -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -Os -fno-omit-frame-pointer -fno-omit-frame-pointer -mapcs -mno-sched-prolog -mlittle-endian -mabi=apcs-gnu -D__LINUX_ARM_ARCH__=5 -march=armv5te -mtune=arm9tdmi -msoft-float -Uarm -Wdeclaration-after-statement -c -o $(WFLAGS)
 export CFLAGS
 endif
+
+ifeq ($(PLATFORM),SONIX)
+	EXTRA_CFLAGS := -D__KERNEL__ -I$(LINUX_SRC)/include -I$(RT28xx_DIR)/include -mlittle-endian -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -O3 -fno-omit-frame-pointer -fno-optimize-sibling-calls -fno-omit-frame-pointer -mapcs -mno-sched-prolog -mabi=apcs-gnu -mno-thumb-interwork -D__LINUX_ARM_ARCH__=5 -march=armv5te -mtune=arm926ej-s --param max-inline-insns-single=40000  -Uarm -Wdeclaration-after-statement -Wno-pointer-sign -DMODULE $(WFLAGS) 
+	export EXTRA_CFLAGS       
+endif
\ No newline at end of file
diff --git a/MODULE/os/linux/modules.order b/MODULE/os/linux/modules.order
index 92675c5..ed12678 100755
--- a/MODULE/os/linux/modules.order
+++ b/MODULE/os/linux/modules.order
@@ -1 +1 @@
-kernel//home/Ralink/SWRelease_RT3070STA/RT_WIFI_20091120_RT3070STA_release/os/linux/rtnet3070sta.ko
+kernel//home/szricky/work/sdk/snx_sdk/driver/rt3070/MODULE/os/linux/rt3070ap.ko
diff --git a/MODULE/tools/bin2h b/MODULE/tools/bin2h
index d5bf460..83fc78a 100755
Binary files a/MODULE/tools/bin2h and b/MODULE/tools/bin2h differ
diff --git a/Makefile b/Makefile
index a25487b..b0c1790 100755
--- a/Makefile
+++ b/Makefile
@@ -6,6 +6,20 @@ all:
 	$(SHELL) cp_module.sh
 	make -C NETIF/
 
+	mkdir -p bin
+	rm -rf bin/*.ko
+
+	#--------------------------------------------------
+	# Ricky: strip all  files but can't work on board?
+	#        As I check the image from Sonix, all the modules didn't striped.
+	arm-linux-strip ./UTIL/os/linux/rtutil3070ap.ko
+	arm-linux-strip ./NETIF/os/linux/rtnet3070ap.ko
+	arm-linux-strip ./MODULE/os/linux/rt3070ap.ko
+
+	cp ./UTIL/os/linux/rtutil3070ap.ko bin/
+	cp ./NETIF/os/linux/rtnet3070ap.ko bin/
+	cp ./MODULE/os/linux/rt3070ap.ko   bin/
+
 clean:
 	make -C UTIL/ clean
 	make -C MODULE/ clean
diff --git a/NETIF/Makefile b/NETIF/Makefile
index 32ef9aa..d02f1ba 100755
--- a/NETIF/Makefile
+++ b/NETIF/Makefile
@@ -15,7 +15,7 @@ endif
 RTMP_SRC_DIR = $(RT28xx_DIR)/RT$(CHIPSET)
 
 #PLATFORM: Target platform
-PLATFORM = PC
+#PLATFORM = PC
 #PLATFORM = 5VT
 #PLATFORM = IKANOS_V160
 #PLATFORM = IKANOS_V180
@@ -37,6 +37,7 @@ PLATFORM = PC
 #PLATFORM = KODAK_DC
 #PLATFORM = DM6446
 #PLATFORM = FREESCALE8377
+PLATFORM = SONIX
 
 #RELEASE Package
 RELEASE = OSABL_UTIL
@@ -53,6 +54,11 @@ ifeq ($(TARGET),THREADX)
 MAKE = gmake
 endif
 
+ifeq ($(PLATFORM), SONIX)
+LINUX_SRC = /home/szricky/work/sdk/snx_sdk/kernel/linux-2.6.35.12/src
+CROSS_COMPILE = arm-linux-
+endif
+
 ifeq ($(PLATFORM),5VT)
 LINUX_SRC = /project/stable/5vt/ralink-2860-sdk/linux-2.6.17
 CROSS_COMPILE = /opt/crosstool/uClibc_v5te_le_gcc_4_1_1/bin/arm-linux-
diff --git a/NETIF/os/linux/config.mk b/NETIF/os/linux/config.mk
index 98f0c8f..2466916 100755
--- a/NETIF/os/linux/config.mk
+++ b/NETIF/os/linux/config.mk
@@ -565,3 +565,8 @@ ifeq ($(PLATFORM),DM6446)
 	CFLAGS := -nostdinc -iwithprefix include -D__KERNEL__ -I$(RT28xx_DIR)/include -I$(LINUX_SRC)/include  -Wall -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -Os -fno-omit-frame-pointer -fno-omit-frame-pointer -mapcs -mno-sched-prolog -mlittle-endian -mabi=apcs-gnu -D__LINUX_ARM_ARCH__=5 -march=armv5te -mtune=arm9tdmi -msoft-float -Uarm -Wdeclaration-after-statement -c -o $(WFLAGS)
 export CFLAGS
 endif
+
+ifeq ($(PLATFORM),SONIX)
+	EXTRA_CFLAGS := -D__KERNEL__ -I$(LINUX_SRC)/include -I$(RT28xx_DIR)/include -mlittle-endian -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -O3 -fno-omit-frame-pointer -fno-optimize-sibling-calls -fno-omit-frame-pointer -mapcs -mno-sched-prolog -mabi=apcs-gnu -mno-thumb-interwork -D__LINUX_ARM_ARCH__=5 -march=armv5te -mtune=arm926ej-s --param max-inline-insns-single=40000  -Uarm -Wdeclaration-after-statement -Wno-pointer-sign -DMODULE $(WFLAGS) 
+	export EXTRA_CFLAGS
+endif
\ No newline at end of file
diff --git a/NETIF/os/linux/usb_main_dev.c b/NETIF/os/linux/usb_main_dev.c
index f712c3b..0142d04 100755
--- a/NETIF/os/linux/usb_main_dev.c
+++ b/NETIF/os/linux/usb_main_dev.c
@@ -20,7 +20,7 @@
 // Otherwise post to forum in ralinktech's web site(www.ralinktech.com) and let all users help you. ***
 MODULE_AUTHOR("Paul Lin <paul_lin@ralinktech.com>");
 MODULE_DESCRIPTION("RT2870 Wireless Lan Linux Driver");
-
+MODULE_LICENSE("GPL");
 
 extern USB_DEVICE_ID rtusb_dev_id[];
 extern INT const rtusb_usb_id_len;
diff --git a/UTIL/Makefile b/UTIL/Makefile
index fb61c32..3842622 100755
--- a/UTIL/Makefile
+++ b/UTIL/Makefile
@@ -15,7 +15,7 @@ endif
 RTMP_SRC_DIR = $(RT28xx_DIR)/RT$(CHIPSET)
 
 #PLATFORM: Target platform
-PLATFORM = PC
+#PLATFORM = PC
 #PLATFORM = 5VT
 #PLATFORM = IKANOS_V160
 #PLATFORM = IKANOS_V180
@@ -37,6 +37,7 @@ PLATFORM = PC
 #PLATFORM = KODAK_DC
 #PLATFORM = DM6446
 #PLATFORM = FREESCALE8377
+PLATFORM = SONIX
 
 #RELEASE Package
 RELEASE = OSABL_UTIL
@@ -53,6 +54,11 @@ ifeq ($(TARGET),THREADX)
 MAKE = gmake
 endif
 
+ifeq ($(PLATFORM), SONIX)
+LINUX_SRC = /home/szricky/work/sdk/snx_sdk/kernel/linux-2.6.35.12/src
+CROSS_COMPILE = arm-linux-
+endif
+
 ifeq ($(PLATFORM),5VT)
 LINUX_SRC = /project/stable/5vt/ralink-2860-sdk/linux-2.6.17
 CROSS_COMPILE = /opt/crosstool/uClibc_v5te_le_gcc_4_1_1/bin/arm-linux-
diff --git a/UTIL/os/linux/config.mk b/UTIL/os/linux/config.mk
index 98f0c8f..68b275f 100755
--- a/UTIL/os/linux/config.mk
+++ b/UTIL/os/linux/config.mk
@@ -469,7 +469,6 @@ endif
 
 ifeq ($(PLATFORM),5VT)
 CFLAGS := -D__KERNEL__ -I$(LINUX_SRC)/include -I$(RT28xx_DIR)/include -mlittle-endian -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -O3 -fno-omit-frame-pointer -fno-optimize-sibling-calls -fno-omit-frame-pointer -mapcs -mno-sched-prolog -mabi=apcs-gnu -mno-thumb-interwork -D__LINUX_ARM_ARCH__=5 -march=armv5te -mtune=arm926ej-s --param max-inline-insns-single=40000  -Uarm -Wdeclaration-after-statement -Wno-pointer-sign -DMODULE $(WFLAGS) 
-
 export CFLAGS
 endif
 
@@ -565,3 +564,8 @@ ifeq ($(PLATFORM),DM6446)
 	CFLAGS := -nostdinc -iwithprefix include -D__KERNEL__ -I$(RT28xx_DIR)/include -I$(LINUX_SRC)/include  -Wall -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -Os -fno-omit-frame-pointer -fno-omit-frame-pointer -mapcs -mno-sched-prolog -mlittle-endian -mabi=apcs-gnu -D__LINUX_ARM_ARCH__=5 -march=armv5te -mtune=arm9tdmi -msoft-float -Uarm -Wdeclaration-after-statement -c -o $(WFLAGS)
 export CFLAGS
 endif
+
+ifeq ($(PLATFORM), SONIX)
+EXTRA_CFLAGS := -D__KERNEL__ -I$(LINUX_SRC)/include -I$(RT28xx_DIR)/include -mlittle-endian -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -O3 -fno-omit-frame-pointer -fno-optimize-sibling-calls -fno-omit-frame-pointer -mapcs -mno-sched-prolog -mabi=apcs-gnu -mno-thumb-interwork -D__LINUX_ARM_ARCH__=5 -march=armv5te -mtune=arm926ej-s --param max-inline-insns-single=40000  -Uarm -Wdeclaration-after-statement -Wno-pointer-sign -DMODULE $(WFLAGS) 
+export EXTRA_CFLAGS
+endif
diff --git a/UTIL/os/linux/rt_usb_util.c b/UTIL/os/linux/rt_usb_util.c
index bc262b8..8afb8ab 100755
--- a/UTIL/os/linux/rt_usb_util.c
+++ b/UTIL/os/linux/rt_usb_util.c
@@ -123,7 +123,11 @@ void *rausb_buffer_alloc(struct usb_device *dev,
 							gfp_t mem_flags,
 							dma_addr_t *dma)
 {
-	return usb_buffer_alloc(dev, size, mem_flags, dma);
+	
+	//return usb_buffer_alloc(dev, size, mem_flags, dma);
+	
+	// fixed for Linux-2.6.xx
+	return usb_alloc_coherent(dev, size, mem_flags, dma);
 }
 EXPORT_SYMBOL(rausb_buffer_alloc);
 
@@ -133,7 +137,10 @@ void rausb_buffer_free(struct usb_device *dev,
 							void *addr,
 							dma_addr_t dma)
 {
-	usb_buffer_free(dev, size, addr, dma);
+	//usb_buffer_free(dev, size, addr, dma);
+	
+	// fixed for Linux-2.6.xx
+	usb_free_coherent(dev, size, addr, dma);
 }
 EXPORT_SYMBOL(rausb_buffer_free);
 #endif // LINUX_VERSION_CODE //
